### DISCUSSION TITLE
MkDocs

### DISCUSSION THREAD

**Discussion by loleg** [cividi/handbook#2] [Ideas]
I've added some thoughts about the process to the [wiki here](https://github.com/cividi/handbook/wiki) and plan to install https://github.com/mkdocs into this repo. Why MkDocs and not..

- [Pelican](https://github.com/getpelican/pelican), like [ogd-handbook-site](https://github.com/opendata-swiss/ogd-handbook-site)
- [Gridsome](https://gridsome.org/), like [our blog](https://cividi.be)
- [ReadTheDocs](https://readthedocs.io) or [GitBook](https://www.gitbook.com)

The question is really, why not? MkDocs is open source, fully self hostable and Python based, seems to be actively supported and has a lot of well maintained plugins. I haven't used it yet in a project, and am keen to try to set up something usable. We won't lose any of the content anyway, as having the (Markdown) content in this repo is a pre-requisite. Other questions that needs to be answered in this test:

- Is it searchable? Multilingual? Support annotations or other cool doc features?
- Does it have good theme support? Could we easily make it fit our CI/UX?
- Can it be simply built and hosted (e.g. at Vercel or GitHub Pages/Actions)?
- Can it also easily include [source code-level](https://medium.com/@peterkong/comparison-of-python-documentation-generators-660203ca3804) documentation?

**Comment by loleg** [comment]
@bkarolina @n0rdlicht I've added an initial fully working project structure along with a couple of further thoughts in this issue.

**Reply by n0rdlicht** [reply]
Thanks @loleg. I converted it into a discussion thread to talk about the criteria.

**Comment by n0rdlicht** [comment]
Regarding hosting: Let's give vercel another round here.

**Comment by n0rdlicht** [comment]
Another option we could consider: [VuePress](https://vuepress.vuejs.org)

Pros:
- supports multilanguage
- built in search, integrations with [Algolia](https://community.algolia.com/docsearch/) search if needed
- statically publishes `.md` files based on a `.vuepress/config.js` as `html/css/js` assets
- easily themable, nice built in theme with color and logo configuration options

Cons:
- no built in way to generate documentation from source-code documentation automatically

**Comment by n0rdlicht** [comment]
Deployment is now setup via GitHub Actions to deploy on pushes / merges to `main`, linked to [handbook.cividi.space](https://handbook.cividi.space).

**Reply by loleg** [reply]
Woohoo! The pencil icon link doesn't work yet though..
